server/src/modules/campaign-monitor/campaign-monitor.service.ts

```diff
@@ -1665,7 +1735,7 @@ export default class CampaignMonitorService {
 									}
 
 									${
-										get(upsertFails, '0') || upsertFails.length
+										get(upsertFails, '0') || upsertFails?.length
```

Comment:
why do you check on both the first element existing and the array having a length?
you can just check only for the first element:
```suggestion
										upsertFails?.[0]
```
unless you're worried that items in the upsertFails array can be undefined or null?